gax-java/pom.xml

```diff
@@ -158,6 +158,13 @@
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-bom</artifactId>
+        <version>1.34.1</version>
```

Comment:
I think we already have it in the [shared-dependencies](https://github.com/googleapis/sdk-platform-java/blob/1463d645edf96087335618542f7bbca722a97b71/java-shared-dependencies/third-party-dependencies/pom.xml#L41), can we try to remove it from there?

Reply:
Ah yeah, let me try and remove.

Reply:
It seems like only two jars were added in shared-deps: https://github.com/googleapis/sdk-platform-java/blob/1463d645edf96087335618542f7bbca722a97b71/java-shared-dependencies/third-party-dependencies/pom.xml#L185-L194

CC: @JoeWang1127 IIRC you mentioned this to me before. What was the reason for only these and not the BOM?

Reply:
Hmm, also. It seems the the pom for Gax flows like Gax -> Gax-Parent -> Generator-Pom-Parent -> Shared-Configs. I might be missing where it's pulling in shared-deps (or does it?)

Reply:
> What was the reason for only these and not the BOM?

I thought only these two artifacts are needed.

> It seems the the pom for Gax flows like Gax -> Gax-Parent -> Generator-Pom-Parent -> Shared-Configs

I think this is right, everything you add to the gax, it's eventually be part of shared-dependencies.

Reply:
> I thought only these two artifacts are needed.

Gotcha. I'm looking to add `opentelemetry-sdk-testing` with test scope for now and I think in the future we'll need to add `opentelemetry-sdk`. Would you have an objections if update shared-deps to pull in the BOM?

> I think this is right, everything you add to the gax, it's eventually be part of shared-dependencies.

If I'm understanding this correctly, this means that we'll need to add it to both gax and shared-deps, right? They should have the same version as renovate-bot should update them together?

Reply:
We should not need `opentelemetry-sdk`, the only thing we need should be `opentelemetry-api`. Is there a code reference to anything other than `opentelemetry-api`?

Reply:
Discussed offline. We don't need to create the OpenTelemetry client so no need to have the `opentelemetry-sdk` package.